# Actividad 1

Alumno: Víctor David Silva

Cohorte: 14

Actividad 1: Redactar un pequeño informe, justificando cada uno de los pasos que se
plantean y las conclusiones que se obtienen.
1) Elegir una serie univariables, pueden tomarla del repositorio GitHub [https://github.com/jbrownlee/Datasets.git] que se compartió en la clase 2, de otro repositorio que indiquen o series propias con las que ya trabajen. Describir todos los datos que considere relevantes para un análisis.
2) Descomponer dicha serie usando un modelo ETS aditivo, y comparar la tendencia que se obtiene con el filtro de Hodrick-Prescott y realizar modelos de suavizado exponencial de 1ero, 2ndo y tercer orden. Calcular las funciones ACF y PACF.
3) Hacer la autocorrelación para distintos valores de rezago, hacer el correlograma.


### Actividad 1.1

El dataset elegido es monthly-car-sales.

In [1]:
import pandas as pd

In [2]:
# url del dataset
url = 'https://raw.githubusercontent.com/jbrownlee/Datasets/master/monthly-car-sales.csv'

# Crear un dataframe de la url
df = pd.read_csv(url)

In [3]:
# Visualizar el dataframe
df

Month  Sales
0    1960-01   6550
1    1960-02   8728
2    1960-03  12026
3    1960-04  14395
4    1960-05  14587
..       ...    ...
103  1968-08  16722
104  1968-09  14385
105  1968-10  21342
106  1968-11  17180
107  1968-12  14577

[108 rows x 2 columns]

Tiene 108 filas, con 2 columnas, una de ellas el mes y año, y la otra las ventas.

In [4]:
# Infor del dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 108 entries, 0 to 107
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Month   108 non-null    object
 1   Sales   108 non-null    int64 
dtypes: int64(1), object(1)
memory usage: 1.8+ KB


No hay datos faltantes y se observa el tipo de dato de cada columna.

In [5]:
# Descripción del dataframe
df.describe()

Sales
count    108.000000
mean   14595.111111
std     4525.213913
min     5568.000000
25%    11391.250000
50%    14076.000000
75%    17595.750000
max    26099.000000

Se observan la cantidad de filas, la media de estas, la desviación estandar, el valor mínimo, el valor méximo y los cuartiles.

### Actividad 1.2

In [6]:
# Grafica del dataframe
df.plot()

ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.

In [9]:
# Descomponer usando un modelo ETS aditivo
from statsmodels.tsa.seasonal import seasonal_decompose

# Con función seasonal_decompose, usando de modelo aditivo, separar la columna en tendencia, seasonal y resid.
result = seasonal_decompose(df['Sales'], model = 'additive')
result.plot()

ValueError: You must specify a period or x must be a pandas object with a PeriodIndex or a DatetimeIndex with a freq not set to None